test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,769 @@
+            }
+        } else {
+            if (introduced.introducedStable == null) {
+                introduced.introducedStable = version;
+            }
+        }
+    }
+
+    private boolean isPreview(Element el, String uniqueId, String currentVersion) {
+        while (el != null) {
+            Symbol s = (Symbol) el;
+            if ((s.flags() & Flags.PREVIEW_API) != 0) {
+                return true;
+            }
+            el = el.getEnclosingElement();
+        }
+
+        return LEGACY_PREVIEW_METHODS.containsKey(currentVersion)
+                && LEGACY_PREVIEW_METHODS.get(currentVersion).contains(uniqueId);
+    }
+
+    private void testThisModule(String moduleName) throws Exception {
+        Path home = Paths.get(System.getProperty("java.home"));
+        Path srcZip = home.resolve("lib").resolve("src.zip");
+        if (Files.notExists(srcZip)) {
+            //possibly running over an exploded JDK build, attempt to find a
+            //co-located full JDK image with src.zip:
+            Path testJdk = Paths.get(System.getProperty("test.jdk"));
+            srcZip = testJdk.getParent().resolve("images").resolve("jdk").resolve("lib").resolve("src.zip");
+        }
+        File f = new File(srcZip.toUri());
+        if (!f.exists() && !f.isDirectory()) {
+            throw new SkippedException("Skipping Test because src.zip wasn't found");
+        }
+        if (Files.isReadable(srcZip)) {
+            URI uri = URI.create("jar:" + srcZip.toUri());
+            try (FileSystem zipFO = FileSystems.newFileSystem(uri, Collections.emptyMap())) {
+                Path root = zipFO.getRootDirectories().iterator().next();
+                Path packagePath = root.resolve(moduleName);
+                try (StandardJavaFileManager fm =
+                             tool.getStandardFileManager(null, null, null)) {
+                    JavacTask ct = (JavacTask) tool.getTask(null,
+                            fm,
+                            null,
+                            List.of("--add-modules", moduleName, "-d", "."),
+                            null,
+                            Collections.singletonList(SimpleJavaFileObject.forSource(URI.create("myfo:/Test.java"), "")));
+                    ct.analyze();
+                    Elements elements = ct.getElements();
+                    elements.getModuleElement("java.base");
+                    try (EffectiveSourceSinceHelper javadocHelper = EffectiveSourceSinceHelper.create(ct, List.of(root), this)) {
+                        processModuleCheck(elements.getModuleElement(moduleName), ct, packagePath, javadocHelper);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        wrongTagsList.add("Initiating javadocHelperFailed " + e.getMessage());
+                    }
+                    if (!wrongTagsList.isEmpty()) {
+                        throw new Exception(wrongTagsList.toString());
```

Comment:
This looks like it might be the exception to end the test, right, and the wrongTags could be long.
I'd suggest writing a heading to `System.err`, then`wrongTagsList.forEach(System.err::println);` and then throw the exception with a summary string, like `throw new Exception(wrongTagsList.size() + " problems found");

Reply:
Done. Added in 32edb4d811978f87bd2f850c214191e8fc7b8c88